plugins/reindexer/dao.go

```diff
@@ -50,17 +50,23 @@ func reindex(ctx context.Context, sourceIndex string, config *reindexConfig, wai
 
 	// If mappings are not passed, we fetch the mappings of the old index.
 	if config.Mappings == nil {
-		config.Mappings, err = mappingsOf(ctx, sourceIndex)
-		if err != nil {
-			return nil, fmt.Errorf(`error fetching mappings of index "%s": %v`, sourceIndex, err)
+		_, found := util.Find(config.Action, "mappings")
```

Comment:
@ShahAnuj2610 you can add const for "data", "mappings"...
same as https://github.com/appbaseio/arc/blob/dev/util/plans.go#L11